### PR TITLE
Add UpdatedTypes to WatchHotReloadService.Update

### PR DESCRIPTION
**Pull request by tmat**
Fixes https://github.com/dotnet/roslyn/issues/54336

### CODE REVIEW

**Comment by tmat** [comment]
@davidwengier @pranavkm

**Review by pranavkm — APPROVED** [review]
That was easy!

File: src/Features/Core/Portable/EditAndContinue/EditSession.cs [hunk 1]
```diff
@@ -762,7 +761,7 @@ public async ValueTask<SolutionUpdate> EmitSolutionUpdateAsync(Solution solution
                             pdbStream.ToImmutableArray(),
                             projectChanges.LineChanges,
                             updatedMethodTokens,
-                            updatedTypes: ImmutableArray<int>.Empty,
+                            updatedTypeTokens,
```
Comment (davidwengier) [inline src/Features/Core/Portable/EditAndContinue/EditSession.cs:764]: I did this in main-vs-deps a while ago.. did this get lost in a merge?
https://github.com/dotnet/roslyn/blob/main-vs-deps/src/Features/Core/Portable/EditAndContinue/EditSession.cs#L764

**Review by davidwengier — APPROVED** [review]

File: src/Features/Core/Portable/EditAndContinue/EditSession.cs (hunk 1 above)
Comment (tmat) [reply src/Features/Core/Portable/EditAndContinue/EditSession.cs:764]: IDK :(

**Pull request merged**